udata/tests/dataset/test_dataset_model.py

```diff
@@ -59,6 +60,21 @@ def test_add_resource_without_checksum(self):
         assert len(dataset.resources) == 2
         assert dataset.resources[0].id == resource.id
 
+    def test_add_two_resources_with_same_id(self):
+        uuid = uuid4()
+        user = UserFactory()
+        dataset = DatasetFactory(owner=user)
+        resource_a = ResourceFactory(id=uuid)
+        resource_b = ResourceFactory(id=uuid)
+
+        dataset.add_resource(resource_a)
+        dataset.add_resource(ResourceFactory())
+        try:
+            dataset.add_resource(resource_b)
+            assert False
+        except RuntimeError:
+            print("Ok!")
```

Comment:
```suggestion
        with pytest.raises(RuntimeError):
            dataset.add_resource(resource_b)
```

Reply:
Changed in 35e775e1186ade0b607a3610fa251d262a8f487e